### PR TITLE
Add bug report issue template

### PR DESCRIPTION
**Pull request by NotMoni**
Since we are moving over to v3.

We should bring some consistency and help **and** make it easier for people to report bugs. Most of the time, ppl don't report them because they can't find a way or they are too lazy to type out every detail. So with this template, we can reduce work for the bug catchers and it will be easier for us to find the bug.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Your Environment**
+
+<!--- Include as many relevant details about the environment you experienced the bug in -->
+
+ | software         | version
+| ---------------- | -------
+| node-fetch          |
```
Comment (MitMaro) [inline .github/ISSUE_TEMPLATE/bug_report.md:31]: Nitpick: But it seems there was some effort in aligning these, should these also be aligned?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+**To Reproduce**
+Steps to reproduce the behavior:
+1. 
+2. 
+3. 
+4. 
+
+**Expected behavior**
+<!--- If you're suggesting a change/improvement, tell us how it should work -->
```
Comment (MitMaro) [inline .github/ISSUE_TEMPLATE/bug_report.md:19]: Should there be a separate issue template for changes/improvements?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (NotMoni) [reply .github/ISSUE_TEMPLATE/bug_report.md:31]: It won't actually affect it, when its rendered I think.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (NotMoni) [reply .github/ISSUE_TEMPLATE/bug_report.md:19]: check out #816

**Comment by NotMoni** [comment]
> The directions under "Expected behavior" are html comments, but other directions are not. Should other directions also be comments?

I don't see any HTML comments.  🤔 ?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (MitMaro) [reply .github/ISSUE_TEMPLATE/bug_report.md:31]: Yup, the render would be the same either way. :)

**Comment by MitMaro** [comment]
> > The directions under "Expected behavior" are html comments, but other directions are not. Should other directions also be comments?
> 
> I don't see any HTML comments.  ?

For example the line `<!--- If you're suggesting a change/improvement, tell us how it should work -->`

Since Markdown doesn't really provide a comment mechanism, they are generally used as a workaround for adding lines to a markdown file that should not be rendered when it is rendered as HTML.

Having the instructions as comments would mean that if the user doesn't remove the instructions, they would not be shown in the final output.

ESLint is the one project that I can think of that does something similar: https://raw.githubusercontent.com/eslint/eslint/3a5fbb3d634be688615950c0a5fa8aead6ff08b5/.github/ISSUE_TEMPLATE/BUG_REPORT.md

**Comment by NotMoni** [comment]
Yep, I see that. It the file is .md so it shouldn't matter? and should it be good?

**Comment by NotMoni** [comment]
What do you suggest as an alternative?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+<!--- If fetch is behaving incorrectly, please include a link to the spec or documentation that supports that your expected behavior is correct. -->
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
```
Comment (MitMaro) [inline .github/ISSUE_TEMPLATE/bug_report.md:23]:
```suggestion
<!--- If applicable, add screenshots to help explain your problem. -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+name: 🐞 Bug report
+about: Create a report to help us improve node-fetch
+title: "[BUG] "
+labels: bug
+---
+
+**Describe the bug**
+A clear and concise description of what the bug is.
```
Comment (MitMaro) [inline .github/ISSUE_TEMPLATE/bug_report.md:9]:
```suggestion
<!--- A clear and concise description of what the bug is. -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+ | software         | version
+| ---------------- | -------
+| node-fetch          |
+| node             |
+| npm              |
+| Operating System |
+
+**Additional context**
+Add any other context about the problem here.
```
Comment (MitMaro) [inline .github/ISSUE_TEMPLATE/bug_report.md:37]:
```suggestion
<!--- Add any other context about the problem here. -->
```

**Pull request merged**